astropy/timeseries/periodograms/base.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from astropy.timeseries import TimeSeries, BinnedTimeSeries
+
+__all__ = ['BasePeriodogram']
+
+
+class BasePeriodogram:
+
```

Comment:
I think you should implement an `__init__` with an `@abstractmethod` decorator that just tells what the initializer is.  That way anyone writing their oen periodogram algorithm knows essentially what `from_timeseries` is going to expect.

Reply:
Added